phpstan.neon.dist

```diff
@@ -0,0 +1,9 @@
+includes:
+    - phpstan-baseline.neon
+
+parameters:
+    level: 3
```

Comment:
level 3 is quite weak. Can't we go at a higher level (possibly using `inferPrivatePropertyTypeFromConstructor` for the 3.x branch where property are not using native types yet) ?

Reply:
The level 4 get `[ERROR] Found 75 errors`.

I plan to bump PHPStan level, but I think it's better
- to have a small pr introducing pr
- to have a pr for level 4
- to have another pr for level 5
- ...

This will give an easier review (and avoid myself to fix everything and getting the pr stale/closed)